pkg/cri/server/service.go

```diff
@@ -104,6 +107,8 @@ type criService struct {
 	// allCaps is the list of the capabilities.
 	// When nil, parsed from CapEff of /proc/self/status.
 	allCaps []string // nolint
+	// a flag value indicates whether the close of CRI service is intended or not.
+	normalClose atomic.Bool
```

Comment:
what about that named by `closed`?

Reply:
I intend to say, it's a "normal close" required by user or upper caller, not something unexpected (e.g: submodule failure). I think `closed` is not sufficient to express the intention.

Reply:
The submodule failure might cause close. But the unexpected close doesn't set the `normalClose`. Is it possible to close it twice?

Reply:
@fuweid No, with the flag value, it will be closed only once whether it's expected or not. 
* CRI service itself will handle both close condition well.
* Containerd will intend to `Close` other services when it meets fatal submodule failure, but from other services' view, it's a requested termination from containerd (in other words, a graceful shutdown).